util/scripts/sync_tas/sync_pi_status_from_tas.py

```diff
@@ -0,0 +1,126 @@
+sys.path.append('/project') 
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "chameleon.settings")
+django.setup()
+logger = logging.getLogger('util.scripts.sync_tas.sync_pi_status')
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from pytas.http import TASClient
+from chameleon.models import PIEligibility
+from django.utils import timezone
+from datetime import datetime
+import argparse
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv
+
+    parser = argparse.ArgumentParser(\
+        description='Import PI eligibility status from TAS')
+    parser.add_argument('--dryrun', type=bool, help=\
+        'Simulate import, no changes applied', default=False)
+    parser.add_argument('--users', type=str, help=\
+        'Synchronize a specific user or set of users using a csv list of usernames', default=None)
+    parser.add_argument('--seteligible', type=bool, help=\
+        'Used with --users flag to set all provided users as PI Eligible', default=False)
+    args = parser.parse_args(argv[1:])
+
+    started = datetime.now()
+    logger.info('Starting... {0}'.format(started))
+    usermodel = get_user_model()
+    if args.users:
+        usernames = args.users.split(',')
+        users = usermodel.objects.filter(username__in=usernames)
+    else:
+        users = usermodel.objects.all()
+
+    if args.users and args.seteligible:
+        updated = set_all_pi_status(args.dryrun, users, "Eligible")
+    else:
+        updated = import_pi_status(args.dryrun, users, TASClient())
+
+    finished = datetime.now()
+    if(args.dryrun):
+        logger.info('Dry run, no updates made')
+
+    ## print summary
+    logger.info('############ Finished ############')
+    for i in updated:
+        logger.info(i)
+    logger.info('{0} users updated'.format(str(len(updated))))
+    logger.info('############ Finished ############')
+    logger.info('Started... {0}'.format(started))
+    logger.info('Finished... {0}'.format(finished))
+    delta = finished - started
+    logger.info('elapsed seconds = {0}'.format(delta.total_seconds()))
+
+def import_pi_status(dryrun=False, users=[], tas=None, set_eligible=False):
+    updated = []
+    if set_eligible:
+        for user in users:
+            save_pi_status(user, "Eligible", dryrun)
+            updated.append((user.username, ["Eligible"]))
+    else:
+        for index, user in enumerate(users):
+            percent_complete = int(index/float(users.count()) * 100)
+            logger.info('{0} of {1}, {2}% complete, next: {3}'.format(\
+                index, users.count(), percent_complete, user.username))
+            if user.pi_eligibility().lower() == 'ineligible': 
+                ''' user is ineligible in portal, let's see if TAS says different '''
+                tas_pi_status = get_tas_pi_status(tas, user.username)
+                if tas_pi_status.lower() != user.pi_eligibility().lower():
```

Comment:
Possible this could wind up setting pi eligibility to an empty string? What possible values are there for this? I thought it was just "eligible" and "ineligible".

Reply:
also `requested`,
Could, but the empty string would have to come from TAS I think, the data looked clean.

Reply:
i think I see what you mean, I think the field is pre-populated with Ineligible by default for all users, if it changed it would be to requested or eligible, but it could be empty if it was set that way by the api and TAS didn't kick it back. I did run this locally and in dev with no empties tho